File: Medium/2025-08-22-3191-FindTheMinimumAreaToCoverAllOnesI/solution.java
class Solution {
    public int minimumArea(int[][] grid) {
        if (grid == null || grid.length == 0) return 0;
        
        int m = grid.length;
        int n = grid[0].length;
        
        // Find the boundaries of all 1's
        int minRow = m, maxRow = -1;
        int minCol = n, maxCol = -1;
        boolean hasOnes = false;
        
        // Scan the matrix to find the boundaries
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] == 1) {
                    hasOnes = true;
                    minRow = Math.min(minRow, i);
                    maxRow = Math.max(maxRow, i);
                    minCol = Math.min(minCol, j);
                    maxCol = Math.max(maxCol, j);
                }
            }
        }
        
        // If no 1's found, return 0
        if (!hasOnes) return 0;
        
        // Calculate area: (maxRow - minRow + 1) * (maxCol - minCol + 1)
        int height = maxRow - minRow + 1;
        int width = maxCol - minCol + 1;
        
        return height * width;
    }
} 

File: Medium/2025-08-22-3191-FindTheMinimumAreaToCoverAllOnesI/solution.py
from typing import List

class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        if not grid or not grid[0]:
            return 0
        
        m, n = len(grid), len(grid[0])
        
        # Find the boundaries of all 1's
        min_row, max_row = m, -1
        min_col, max_col = n, -1
        has_ones = False
        
        # Scan the matrix to find the boundaries
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 1:
                    has_ones = True
                    min_row = min(min_row, i)
                    max_row = max(max_row, i)
                    min_col = min(min_col, j)
                    max_col = max(max_col, j)
        
        # If no 1's found, return 0
        if not has_ones:
            return 0
        
        # Calculate area: (max_row - min_row + 1) * (max_col - min_col + 1)
        height = max_row - min_row + 1
        width = max_col - min_col + 1
        
        return height * width
